### PR TITLE
Upgrade Openresty, activate modules

### PR DESCRIPTION
**Pull request by adragomir**
Some upgrades and cleanup on the Docker container
1. Upgrade to Openresty 1.9.7.3
2. Cleanup docker file image (erase gcc)
3. Enable modules

### CODE REVIEW

**Comment by adragomir** [comment]
ping @ddragosd

File: Dockerfile [hunk 1]
```diff
@@ -85,13 +116,14 @@ RUN  cd /tmp/api-gateway/ \
...
-         INSTALL=/tmp/api-gateway/ngx_openresty-${OPENRESTY_VERSION}/build/install \
+         INSTALL=/tmp/api-gateway/openresty-${OPENRESTY_VERSION}/build/install \
     && echo " ... installing api-gateway-request-validation ..." \
     && tar -xf /tmp/api-gateway/api-gateway-request-validation-${REQUEST_VALIDATION_VERSION}.tar.gz -C /tmp/api-gateway/ \
     && cd /tmp/api-gateway/api-gateway-request-validation-${REQUEST_VALIDATION_VERSION} \
     && make install \
          LUA_LIB_DIR=${_exec_prefix}/api-gateway/lualib \
-         INSTALL=/tmp/api-gateway/ngx_openresty-${OPENRESTY_VERSION}/build/install \
+         INSTALL=/tmp/api-gateway/openresty-${OPENRESTY_VERSION}/build/install \
+    && apk del g++ gcc make \
```
Comment (clehene) [inline Dockerfile:125]: should we instead have a separate build container?

Comment (adragomir) [inline Dockerfile:125]: @clehene good question, I'm really unclear on this pattern of building and running from the separate or different containers. @amuraru had the same idea, I think, but I don't know.

**Comment by ddragosd** [comment]
looks good :+1: thanks @adragomir 

I will run a few tests and then merge it.

File: Dockerfile [hunk 2]
```diff
@@ -48,8 +49,23 @@ RUN  cd /tmp/api-gateway/ \
...
+            --with-http_sub_module \
+            --with-http_dav_module \
+            --with-http_geoip_module \
+            --with-http_gunzip_module  \
+            --with-http_gzip_static_module \
+            --with-http_auth_request_module \
+            --with-http_random_index_module \
+            --with-http_secure_link_module \
+            --with-http_degradation_module \
+            --with-http_auth_request_module  \
```
Comment (ddragosd) [inline Dockerfile:67]: `http_auth_request_module` is duplicated

File: Dockerfile [hunk 3]
```diff
@@ -70,8 +86,23 @@ RUN  cd /tmp/api-gateway/ \
...
```
Comment (ddragosd) [inline Dockerfile:105]: `http_auth_request_module` is duplicated

**Comment by ddragosd** [comment]
I'm integrating this as-is and then I'll make another PR to remove the duplicated modules.

**Comment by ddragosd** [comment]
I've rebased this into `issue-12`. See f298776

**Pull request closed without merging**